From c82490992722fa5435c65c7a830736dc0e47c501 Mon Sep 17 00:00:00 2001
From: Robert Liu <elburrito@google.com>
Date: Thu, 9 Mar 2023 15:50:41 +0000
Subject: [PATCH] Add OPENSSL_EXPORT to BN_mod_inverse_blinded

Change-Id: Ie7543047c6f489ae849b3c27703948f0977c33fd
Reviewed-on: https://boringssl-review.googlesource.com/c/boringssl/+/57905
Commit-Queue: David Benjamin <davidben@google.com>
Reviewed-by: David Benjamin <davidben@google.com>
(cherry picked from commit 8aa51ddfcf1fbf2e5f976762657e21c7aee2f922)
---
 include/openssl/bn.h | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/include/openssl/bn.h b/include/openssl/bn.h
index acde58d29..1934a29d4 100644
--- a/include/openssl/bn.h
+++ b/include/openssl/bn.h
@@ -846,8 +846,9 @@ OPENSSL_EXPORT BIGNUM *BN_mod_inverse(BIGNUM *out, const BIGNUM *a,
 // Note this function may incorrectly report |a| has no inverse if the random
 // blinding value has no inverse. It should only be used when |n| has few
 // non-invertible elements, such as an RSA modulus.
-int BN_mod_inverse_blinded(BIGNUM *out, int *out_no_inverse, const BIGNUM *a,
-                           const BN_MONT_CTX *mont, BN_CTX *ctx);
+OPENSSL_EXPORT int BN_mod_inverse_blinded(BIGNUM *out, int *out_no_inverse,
+                                          const BIGNUM *a,
+                                          const BN_MONT_CTX *mont, BN_CTX *ctx);
 
 // BN_mod_inverse_odd sets |out| equal to |a|^-1, mod |n|. |a| must be
 // non-negative and must be less than |n|. |n| must be odd. This function
